crates/service-client/src/http.rs

```diff
@@ -74,23 +73,28 @@ impl HttpClient {
             .with_native_roots()
             .expect("Can build native roots")
             .https_or_http()
+            .enable_http1()
```

Comment:
Just to be clear, for fixing the actual problem this is the only required change, right? Everything else is just massaging the existing code in different form?

Reply:
It's a mix of this change and the change below where in case of HTTPs we just let ALPN rule the version.